From ba77708a8f28ddba9d25e90d0b8b05553f6dac8c Mon Sep 17 00:00:00 2001
From: "kirk.wang" <kirk.wang@amlogic.com>
Date: Wed, 12 Jul 2023 15:59:51 +0800
Subject: [PATCH 06/10] sc2: sc2_ah212 kernel 5.15 [1/1]

PD#SWPL-131990

Problem:
need birngup sc2_ah212 kernel 5.15

Solution:
add conf for sc2_ah212 kernel 5.15

Verify:
sc2

Signed-off-by: kirk.wang <kirk.wang@amlogic.com>
Change-Id: Ic5e03ae9f47226445fe6deef78b1606686136657
---
 conf/machine/include/mesonsc2_k5.15_64b.inc   | 12 +++
 conf/machine/mesonsc2-ah212-5.15.conf         |  1 +
 .../machine/mesonsc2-ah212-weston-5.15.config | 77 +++++++++++++++++++
 recipes-kernel/linux/5.15/sc2.cfg             | 15 ++++
 4 files changed, 105 insertions(+)
 create mode 100755 conf/machine/include/mesonsc2_k5.15_64b.inc
 create mode 120000 conf/machine/mesonsc2-ah212-5.15.conf
 create mode 100644 conf/machine/mesonsc2-ah212-weston-5.15.config
 create mode 100644 recipes-kernel/linux/5.15/sc2.cfg

diff --git a/conf/machine/include/mesonsc2_k5.15_64b.inc b/conf/machine/include/mesonsc2_k5.15_64b.inc
new file mode 100755
index 00000000..691bf75c
--- /dev/null
+++ b/conf/machine/include/mesonsc2_k5.15_64b.inc
@@ -0,0 +1,12 @@
+# common configuration
+require conf/machine/include/amlogic_common.inc
+require conf/machine/include/amlogic_64b.inc
+
+PREFERRED_VERSION_u-boot = "v2019.%"
+#We can't select revsion in external source mode, but use MACHINE define to control it.
+PREFERRED_VERSION_linux-meson = "5.15.%"
+PREFERRED_VERSION_media-modules = "5.15%"
+PREFERRED_PROVIDER_virtual/gpu ?= "dvalin"
+OVERRIDES .= ":k5.15"
+LINUXLIBCVERSION ?= "5.15%"
+HOSTTOOLS_NONFATAL += " pahole"
diff --git a/conf/machine/mesonsc2-ah212-5.15.conf b/conf/machine/mesonsc2-ah212-5.15.conf
new file mode 120000
index 00000000..d68e20c7
--- /dev/null
+++ b/conf/machine/mesonsc2-ah212-5.15.conf
@@ -0,0 +1 @@
+mesonsc2-ah212-weston-5.15.config
\ No newline at end of file
diff --git a/conf/machine/mesonsc2-ah212-weston-5.15.config b/conf/machine/mesonsc2-ah212-weston-5.15.config
new file mode 100644
index 00000000..85c81b70
--- /dev/null
+++ b/conf/machine/mesonsc2-ah212-weston-5.15.config
@@ -0,0 +1,77 @@
+#@TYPE: Machine
+#@NAME: meson
+
+#@DESCRIPTION: Machine configuration for meson systems
+
+include conf/machine/include/mesonsc2_k5.15_64b.inc
+include conf/machine/include/amlogic_weston.inc
+
+PREFERRED_PROVIDER_virtual/mesa = "libgles-eabihf-dvalin-wayland-drm"
+PREFERRED_PROVIDER_virtual/mesa-gl = "libgles-eabihf-dvalin-wayland-drm"
+PREFERRED_PROVIDER_virtual/libgbm = "libgles-eabihf-dvalin-wayland-drm"
+PREFERRED_PROVIDER_virtual/egl = "libgles-eabihf-dvalin-wayland-drm"
+PREFERRED_PROVIDER_virtual/libgl = "libgles-eabihf-dvalin-wayland-drm"
+PREFERRED_PROVIDER_virtual/libgles1 = "libgles-eabihf-dvalin-wayland-drm"
+PREFERRED_PROVIDER_virtual/libgles2 = "libgles-eabihf-dvalin-wayland-drm"
+
+MACHINE_ESSENTIAL_EXTRA_RRECOMMENDS += " \
+        wififw-qca6174 \
+        qca6174 \
+        tuner-prebuilt \
+        aucpu-fw \
+        gpu \
+        coreutils \
+        "
+
+CHIPSET_NAME = "S905X4"
+PREFERRED_VERSION_qca6174 = "5.15"
+RDEPENDS_packagegroup-amlogic-baserootfs_remove = "aml-subtitleserver"
+RDEPENDS_packagegroup-amlogic-baserootfs_remove = " fuse-exfat"
+KERNEL_DEVICETREE = "sc2_s905x4_ah212_linux.dtb"
+UBOOT_MACHINE = "sc2_ah212_config"
+KERNEL_MODULE_AUTOLOAD += " cxd2856_fe_64"
+KERNEL_MODULE_AUTOLOAD += " snd-soc-dummy_codec"
+KERNEL_MODULE_AUTOLOAD += " snd-soc-aml_t9015"
+KERNEL_MODULE_AUTOLOAD += " meson_ir"
+KERNEL_MODULE_AUTOLOAD += " btqca"
+KERNEL_MODULE_PROBECONF += "btqca"
+KERNEL_MODULE_AUTOLOAD += " hci_uart"
+KERNEL_MODULE_PROBECONF += "hci_uart"
+KERNEL_MODULE_AUTOLOAD += "aml_aucpu"
+KERNEL_MODULE_AUTOLOAD += "dolby_fw"
+KERNEL_MODULE_AUTOLOAD += "meson-cpufreq"
+
+DISTRO_FEATURES_append = " drm"
+DISTRO_FEATURES_append = " gpt-partition"
+DISTRO_FEATURES_append = " kernel_515"
+
+#DISTRO_FEATURES_append = " onepass"
+DISTRO_FEATURES_append = " secure-u-boot optee "
+DISTRO_FEATURES_append = " widevine "
+DISTRO_FEATURES_append = " playready "
+#DISTRO_FEATURES_append = " amlogic-tv "
+DISTRO_FEATURES_append = " amlogic-dvb "
+#DISTRO_FEATURES_remove = " alsa "
+DISTRO_FEATURES_append = " alsa "
+DISTRO_FEATURES_append = " pam"
+DISTRO_FEATURES_BACKFILL_CONSIDERED = "sysvinit"
+DEFAULT_DISTRO_FEATURES += " systemd"
+DISTRO_FEATURES_append = " systemd"
+DISTRO_FEATURES_remove = " sysvinit"
+#DISTRO_FEATURES_append = " opengl"
+DISTRO_FEATURES_append = " gstreamer1"
+DISTRO_FEATURES_append = " aml-dtv "
+#DISTRO_FEATURES_append = " recovery "
+#DISTRO_FEATURES_append = " swupdate "
+DISTRO_FEATURES_append = " aml-libjpeg "
+DISTRO_FEATURES_append = " aml-libvpcodec "
+DISTRO_FEATURES_append = " aml-libvphevcodec "
+DISTRO_FEATURES_append = " gst-plugin-venc "
+
+OVERRIDES .= ":ah212:sc2:onepass"
+MACHINEOVERRIDES .= ":client"
+
+DISTRO_FEATURES_append = " bluetooth bluez5 "
+EXTRA_BLUETOOTH_STUFF := "${@bb.utils.contains('DISTRO_FEATURES', 'bluetooth', bb.utils.contains('DISTRO_FEATURES', 'bluez5', ' bluez5', ' bluez4', d), '', d)}"
+DISTRO_FEATURES_append = "${EXTRA_BLUETOOTH_STUFF}"
+
diff --git a/recipes-kernel/linux/5.15/sc2.cfg b/recipes-kernel/linux/5.15/sc2.cfg
new file mode 100644
index 00000000..8adf506a
--- /dev/null
+++ b/recipes-kernel/linux/5.15/sc2.cfg
@@ -0,0 +1,15 @@
+CONFIG_USERFAULTFD=y
+CONFIG_AUTOFS4_FS=y
+CONFIG_VT=y
+CONFIG_CONSOLE_TRANSLATIONS=y
+CONFIG_HW_CONSOLE=y
+CONFIG_FHANDLE=y
+CONFIG_SERIAL_DEV_BUS=y
+CONFIG_SERIAL_DEV_CTRL_TTYPORT=y
+CONFIG_SYSVIPC=y
+CONFIG_BT_RFCOMM=y
+# CONFIG_CC_HAVE_STACKPROTECTOR_SYSREG is not set
+# CONFIG_BT_HCIBTSDIO is not set
+CONFIG_DRM_MESON_EMULATE_FBDEV=y
+CONFIG_EXFAT_FS=y
+
-- 
2.25.1

